### PR TITLE
net: initialize socket to avoid closing random fd's

### PR DESCRIPTION
**Pull request by theuni**
An excellent spot by @david60.

Even if it isn't causing the fd issue we're looking for, this should be fixed.

### CODE REVIEW

File: src/net.cpp [hunk 1]
```diff
@@ -410,7 +410,7 @@ CNode* CConnman::ConnectNode(CAddress addrConnect, const char *pszDest, bool fCo
 
     // Connect
     bool connected = false;
-    SOCKET hSocket;
+    SOCKET hSocket = INVALID_SOCKET;
```
Comment (Sjors) [inline src/net.cpp]: Why didn't a linter catch this initialized variable?

**Comment by sdaftuar** [comment]
utACK

**Comment by promag** [comment]
utACK 62e07f0.

**Comment by Sjors** [comment]
I'm currently running `git bisect` to figure out where the crashes started. Once that's done I'll also check if this commit makes them go away.

**Comment by laanwj** [comment]
utACK https://github.com/bitcoin/bitcoin/pull/12326/commits/96dbd381cf0ded169406bab3b1ba911a13d563c5 - this is obviously correct. Hope it solves the issue too.

**Pull request merged**